### PR TITLE
Document highlights on async/await keywords should highlight other oc…

### LINKED ISSUE #22944: Document highlights on async/await keywords should highlight other occurrences in the same body
```ts
[|async|] function foo() {
  let x = [|await|] 100;
  let y = [|await|] 200;
  return 300
}
```

I'd expect each document highlight above to contain the others.

I wouldn't think the `return` keyword would be included here.

### PR DESCRIPTION
**Pull request by Zzzen**
…currences in the same body fix #22944

Fixes https://github.com/Microsoft/TypeScript/issues/22944

### CODE REVIEW

File: tests/cases/fourslash/getOccurrencesAsyncAwait3.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// <reference path='fourslash.ts' />
+
+// Not valid TS ('await' expression is only allowed within an async function.)
+
+////a/**/wait 100;
+////async function f() {
+////    await 300;
+////}
+
+verify.rangesAreOccurrences(false);
```
Comment (mhegazy) [inline tests/cases/fourslash/getOccurrencesAsyncAwait3.ts]: remove this.

File: src/services/documentHighlights.ts [hunk 2]
```diff
@@ -368,6 +372,35 @@ namespace ts.DocumentHighlights {
...
+        forEachChild(func, aggregate);
+
+        return keywords;
+
+        function aggregate(node: Node): void {
+            if (isAwaitExpression(node)) {
+                pushKeywordIf(keywords, node.getFirstToken(), SyntaxKind.AwaitKeyword);
+            }
+            // Do not cross function boundaries.
+            if (!isFunctionLike(node)) {
```
Comment (mhegazy) [inline src/services/documentHighlights.ts]: `&& !isClassLike(node) && !isInterfaceDeclaration(node) && !isModuleDeclaration(node) && !isTypeAliasDeclaration(node) && !isTypeNode(node)`

**Comment by mhegazy** [comment]
thanks!

**Comment by mhegazy** [comment]
@Zzzen wanna also add support for `yield` in generators?

**Comment by Zzzen** [comment]
haha, i will try

**Pull request merged**